internal/rootcoord/meta_table.go

```diff
@@ -1046,3 +1067,84 @@ func (mt *metaTable) dupMeta() (
 	}
 	return collID2Meta, segID2IndexMeta, indexID2Meta
 }
+
+func (mt *metaTable) AddAlias(collectionAlias string, collectionName string,
+	ts typeutil.Timestamp, ddOpStr func(ts typeutil.Timestamp) (string, error)) error {
+	mt.ddLock.Lock()
+	defer mt.ddLock.Unlock()
+	if _, ok := mt.collAlias2ID[collectionAlias]; ok {
+		return fmt.Errorf("duplicate collection alias, alias = %s", collectionAlias)
+	}
+
+	id, ok := mt.collName2ID[collectionName]
+	if !ok {
+		return fmt.Errorf("aliased collection name does not exist, name = %s", collectionName)
+	}
+	mt.collAlias2ID[collectionAlias] = id
+
+	meta := make(map[string]string)
+	addition := mt.getAdditionKV(ddOpStr, meta)
+	saveAlias := func(ts typeutil.Timestamp) (string, string, error) {
+		k1 := fmt.Sprintf("%s/%s", CollectionAliasMetaPrefix, collectionAlias)
+		v1 := proto.MarshalTextString(&pb.CollectionInfo{ID: id, Schema: &schemapb.CollectionSchema{Name: collectionAlias}})
+		meta[k1] = v1
+		return k1, v1, nil
+	}
+
+	err := mt.client.MultiSave(meta, ts, addition, saveAlias)
+	if err != nil {
+		log.Error("SnapShotKV MultiSave fail", zap.Error(err))
+		panic("SnapShotKV MultiSave fail")
+	}
+	return nil
+}
+
+func (mt *metaTable) DeleteAlias(collectionAlias string, ts typeutil.Timestamp, ddOpStr func(ts typeutil.Timestamp) (string, error)) error {
+	mt.ddLock.Lock()
```

Comment:
same, what if collection not exist?

Reply:
I have checked whether if the alias exists. `return fmt.Errorf("alias does not exist, alias = %s", collectionAlias)`